### PR TITLE
Catch DistributedLockTimeoutException in some background processes.

### PR DESCRIPTION
**Pull request by aidmsu**
Catch `DistributedLockTimeoutException` exceptions in `RecurringJobScheduler`, `DelayedJobScheduler`, `ExpirationManager` background processes.

If `DistributedLockTimeoutException` is thrown during running of these processes it doesn't mean that work  wasn't done. It just mean another Hangfire server did the work. Thus we should catch this exception.

In case of a large set of Hangfire servers it will make easier to read logs because they won't contain many records about `DistributedLockTimeoutException`.

### CODE REVIEW

File: src/Hangfire.Core/Server/DelayedJobScheduler.cs [hunk 1]
```diff
@@ -144,39 +145,55 @@ public override string ToString()
         private bool EnqueueNextScheduledJob(BackgroundProcessContext context)
         {
             using (var connection = context.Storage.GetConnection())
-            using (connection.AcquireDistributedLock("locks:schedulepoller", DefaultLockTimeout))
```
Comment (odinserj) [inline src/Hangfire.Core/Server/DelayedJobScheduler.cs]: Why did you remove the `using` statement? Why not to wrap the whole block with the `try/catch` statement?

File: src/Hangfire.Core/Server/RecurringJobScheduler.cs [hunk 2]
```diff
@@ -115,33 +115,50 @@ public void Execute(BackgroundProcessContext context)
             _throttler.Throttle(context.CancellationToken);
 
             using (var connection = context.Storage.GetConnection())
-            using (connection.AcquireDistributedLock("recurring-jobs:lock", LockTimeout))
```
Comment (odinserj) [inline src/Hangfire.Core/Server/RecurringJobScheduler.cs]: The same question about the `using` block.

File: src/Hangfire.SqlServer/ExpirationManager.cs [hunk 3]
```diff
@@ -67,7 +68,18 @@ public void Execute(CancellationToken cancellationToken)
 
                 _storage.UseConnection(connection =>
                 {
-                    SqlServerDistributedLock.Acquire(connection, DistributedLockKey, DefaultLockTimeout);
```
Comment (odinserj) [inline src/Hangfire.SqlServer/ExpirationManager.cs]: Why not to wrap the whole `UseConnection` method with the `try/catch` block?

File: src/Hangfire.Core/Server/DelayedJobScheduler.cs (hunk 1 above)
Comment (aidmsu) [inline src/Hangfire.Core/Server/DelayedJobScheduler.cs]: @odinserj  I'd like to show that `AcquireDistributedLock` can throw `DistributedLockTimeoutException` but not whole block.

**Comment by aidmsu** [comment]
@odinserj I added logging timeout exceptions with the debug level.

**Comment by unicrus** [comment]
Guys, please merge it.

**Comment by odinserj** [comment]
Merged into `master` via 1d6ad9ae50d4011863dd5eba905ac2ad365ab979, 7f521e97ed8246895ef7fba7292445b541d5f0fc, 7c141c5f5c972f4069c1ff1d0c167535b442f64b, 5aed50a41d70c1b7d46cad7c2c49797e1a32b1a8, fecf74a1ff5dd90bb356980b0b7966863ea6a2a8 commits.
